src/backend/storeManagers/hyperplay/games.ts

```diff
@@ -458,6 +461,49 @@ async function getAccessCodeGatedPlatforms(
   return validateResult.platforms
 }
 
+async function getTokenGatedPlatforms(channel_id: number): Promise<PlatformsMetaInterface> {
+  if (!window.ethereum) throw 'Window.ethereum provider not found'
+  const provider = new ethers.BrowserProvider(window.ethereum)
+  const signer = await provider.getSigner()
+  const address = await signer.getAddress()
+
+  const domain = window.location.host
+  const origin = window.location.origin
+
+  const statementRes = await axios.get(
+    valistBaseApiUrlv1 + '/license_contracts/validate/get-nonce'
+  )
+  const statement = String(statementRes?.data)
+
+  const siweMessage = new SiweMessage({
```

Comment:
**Optional**, but same as above we can have these as helper functions, this would make it easier if we need to utilize in multiple places

Can consider to chance the below `chain` as param too.

```typescript
async function createSiweMessage(signerAddress: string): Promise<SiweMessage> {
  const domain = window.location.host;
  const origin = window.location.origin;

  const statementRes = await axios.get(`${BASE_API_URL}/license_contracts/validate/get-nonce`);
  const statement = String(statementRes?.data);

  return new SiweMessage({
    domain,
    address: signerAddress,
    statement,
    uri: origin,
    version: '1',
    chainId: 1,
  });
}
```

Then we can easily use all this as 

```typescript
async function validateLicense(signer: ethers.Signer, channelId: number): Promise<PlatformsMetaInterface> {
  const address = await signer.getAddress();
  const siweMessage = await createSiweMessage(address);
  const message = siweMessage.prepareMessage();
  const signature = await signer.signMessage(message);

  const validateResult = (await axios.post<LicenseConfigValidateResult>(`${BASE_API_URL}/license_contracts/validate`, {
    message,
    signature,
    address,
    channel_id: channelId,
  })).data;

  if (validateResult.valid !== true) {
    throw new Error(`Address ${address} is not valid for channel id ${channelId}`);
  }

  if (!validateResult.platforms) {
    throw new Error('Token gated platforms returned by the validate url were undefined');
  }

  return validateResult.platforms;
}

async function getTokenGatedPlatforms(channelId: number): Promise<PlatformsMetaInterface> {
  const signer = await getSigner();
  return await validateLicense(signer, channelId);
}
```
You can move things accordingly as well, that is just an example the above

Reply:
Awesome suggestion, I broke up the SIWE logic and moved some stuff around!